PhysicsTools/PatAlgos/plugins/PATSecondaryVertexSlimmer.cc

```diff
@@ -32,6 +37,7 @@ namespace pat {
 pat::PATSecondaryVertexSlimmer::PATSecondaryVertexSlimmer(const edm::ParameterSet& iConfig) :
   src_(consumes<reco::VertexCompositePtrCandidateCollection>(iConfig.getParameter<edm::InputTag>("src"))),
   srcLegacy_(mayConsume<std::vector<reco::Vertex> >(iConfig.getParameter<edm::InputTag>("src"))),
+  srcV0s_(mayConsume<reco::VertexCompositeCandidateCollection>(iConfig.getParameter<edm::InputTag>("src"))),
```

Comment:
`mayConsumes` is presently (and may always) trigger prefetching of the item requested. It is possible to determine what type is available before calling `consumes`. This is done by registering a function to call when new products are registered. The registration is done by passing a functor to `callWhenNewProductsRegistered`. The documentation can be found here:
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideEDMGetDataFromEvent#GetterOfProducts